zipline/data/_equities.pyx

```diff
@@ -191,8 +197,13 @@ cpdef _read_bcolz_data(ctable_t table,
 
             for asset in range(nassets):
                 first_row = first_rows[asset]
+                if first_row == -1:
```

Comment:
Yeah you're right...I'm not sure why this isn't getting caught. Looking into it now.

Reply:
Hah, turns out this actually works without the check. Since `first_row`, `last_row`, and `offset` are all -1, we end up trying set the slice `[-1:0]`, which is empty. I much prefer exiting early though, so I'll add the check in both places.